raft/src/main/java/org/apache/kafka/raft/internals/BatchAccumulator.java

```diff
@@ -495,6 +506,10 @@ public int sizeInBytes() {
         public void release() {
             pool.release(initialBuffer);
         }
+
+        public long appendTimestamp() {
+            return data.firstBatch().maxTimestamp();
```

Comment:
This will be the timestamp of the first batch in the MemoryRecords. Do we assume that only a single batch is included in any given MemoryRecords?

Reply:
Yes. I added this check and message to the PR: https://github.com/apache/kafka/pull/10946/files/1964e026b783235e2d9af9c0b116f15ed31e0a5b#diff-b7a2129b03764fbafab69c81e985d8ac6006d55f95307f0e21c70fb4750f61b5R475

Reply:
I think we should document that "we only use the append time of the first batch" here, and append a `to-do` with KAFKA-13020 to make it clear.

Reply:
Thanks @showuon . I added a comment. I didn't add the Jira issue. Apache Kafka tends not link to Jira from the code.

Reply:
@jsancio : for future rerference, it's OK to put Apache JIRA numbers in the code. :)